### PR TITLE
3654 - History: temp

### PR DESCRIPTION
**Pull request by sorja**

### CODE REVIEW

File: src/client/components/Buttons/ButtonHistory/ButtonHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { useToggleHistory } from 'client/components/Buttons/ButtonHistory/hooks/useToggleHistory'
+
+type Props = {
+  canEdit: boolean
+  disabled: boolean
+  sectionName: string
+  subSection: string
+  name: string
+  sectionLabelKey: string
+}
```
Comment (minotogna) [inline src/client/components/Buttons/ButtonHistory/ButtonHistory.tsx:20]: Too many props . Most of them can be taken directly through hooks

File: src/client/components/Buttons/ButtonHistory/ButtonHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const historyActive = useHistoryActive()
+  const onClick = useToggleHistory({ name, sectionLabelKey, sectionName, subSection })
+
+  const history = useHistory()
+  // Show toggle button when browsing history for current section
+  const currentSectionEnabled = !Objects.isEmpty(
+    history.items?.[Histories.getHistoryItemSectionKey(sectionName, subSection, name)]
+  )
+
+  if (!canEdit && !currentSectionEnabled) return null
```
Comment (minotogna) [inline src/client/components/Buttons/ButtonHistory/ButtonHistory.tsx:36]: This button should be hidden only if description is in read only mode .

File: src/client/components/Buttons/ButtonHistory/ButtonHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  name: string
+  sectionLabelKey: string
+}
+
+const ButtonHistory: React.FC<Props> = (props) => {
+  const { canEdit, disabled, sectionName, subSection, sectionLabelKey, name } = props
+
+  const { t } = useTranslation()
+
+  const historyActive = useHistoryActive()
```
Comment (minotogna) [inline src/client/components/Buttons/ButtonHistory/ButtonHistory.tsx:27]: If this returns a Boolean , better to call useIsHistoryActive

File: src/client/components/Buttons/ButtonHistory/hooks/useToggleHistory.ts [hunk 4]
```diff
@@ -9,18 +9,19 @@ import { DataActions } from 'client/store/data'
 type Returned = () => void
 
 type UseToggleHistoryParams = {
-  name: CommentableDescriptionName
+  name: string
+  sectionLabelKey: string
   sectionName: SectionName
+  subSection: string
```
Comment (minotogna) [inline src/client/components/Buttons/ButtonHistory/hooks/useToggleHistory.ts:15]: What's this?

File: src/client/components/Navigation/History/Items/Item/Item.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+type Props = {
+  datum: ActivityLog<never>
+  sectionItemKey: HistoryItemSectionKey
+}
+
+const _getValue = (activity: ActivityLog<unknown>): CommentableDescriptionValue => {
+  return (activity as ActivityLog<{ description: { value: CommentableDescriptionValue } }>).target.description.value
+}
+
+const useOnClick = () => {
```
Comment (minotogna) [inline src/client/components/Navigation/History/Items/Item/Item.tsx:26]: Better in separate file

File: src/client/components/Buttons/ButtonHistory/hooks/useToggleHistory.ts (hunk 4 above)
Comment (sorja) [reply src/client/components/Buttons/ButtonHistory/hooks/useToggleHistory.ts:15]: eg. descriptions -> it's the state key, should we rename it?
It is used in:
`DataActions.setValue`

**Pull request closed without merging**